enhancements/ibm-public-cloud/ibm-public-cloud-support.md

```diff
@@ -0,0 +1,251 @@
+---
+title: IBM Public Cloud Support
+authors:
+  - "@csrwng"
+reviewers:
+  - "@derekwaynecarr"
+  - "@sttts"
+  - "@deads2k"
+  - "@mfojtik"
+  - "@ironcladlou"
+  - "@crawford"
+  - "@abhinavdahiya"
+  - "@spadgett"
+  - "@miabbott"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2020-02-04
+last-updated: 2020-02-04
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# IBM Public Cloud Support
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+IBM Public Cloud deploys Kubernetes/OpenShift clusters by hosting their control plane
+as pods in a central management cluster. OpenShift v3.11 is currently offered as a managed
+service using this model. This type of setup allows the hosting of many
+cluster control planes on a single Kubernetes cluster. Only the worker nodes are exclusive
+to an individual user cluster and may exist in a different account than the central
+management cluster.
+
+Supporting this type of deployment in OpenShift 4.x requires a different approach to
+installing OpenShift and managing its control plane. Instead of bootstrapping a cluster 
+control plane from scratch on virtual machines, a cluster control plane can be created 
+by applying a set of manifests to an existing cluster in a new namespace. 
+
+Changes are required in different areas of the product in order to make clusters deployed 
+using this method viable. These include changes to the cluster version operator (CVO), web 
+console, second level operators (SLOs) deployed by the CVO, and RHCOS.
+
+## Motivation
+
+Enable OpenShift 4.x deployment on IBM Cloud.
+
+### Goals
+
+- Enable the deployment of OpenShift 4.x on IBM Public Cloud by providing the IBM
+  Public Cloud team the necessary tools to generate manifests needed for a hosted
+  control plane.
+- Ensure that this deployment model remains functional through regular e2e testing
+  on IBM Public Cloud.
+- Make the necessary product changes to make IBM Public Cloud a fully supported 
+  cloud provider.
+- Ensure that clusters running on IBM Public Cloud are conformant OpenShift clusters
+  (pass Kubernetes and OpenShift conformance e2e tests). Fully document any tests that
+  must be skipped due to differences from a traditional deployment.
+
+### Non-Goals
+
+- Make hosted control planes a supported deployment model outside of IBM Public Cloud.
```

Comment:
I find this confusing...IBM Public Cloud means a lot of different things, and a sort of baseline obvious one is having OpenShift support the default "self driving" path in their existing IaaS.  But I guess we're doing hosted control plane *first*?

Maybe the enhancement should be called: "IBM Public Cloud Hosted Control Plane" ?

And one thing I would say here is that we should think of this "fairly" - if some other IaaS showed up and was willing to commit significant resources to maintaining a similar thing...  clearly vast amounts of the design would likely be shared.  But that can come later.

Reply:
Absolutely, I think the things we learn from this work is something we can likely reuse in other cases. And perhaps at the proper time, make the pattern something standalone that's configured per provider. So definitely, this non-goal is a point-in-time statement.

Reply:
> I find this confusing...IBM Public Cloud means a lot of different things, and a sort of baseline obvious one is having OpenShift support the default "self driving" path in their existing IaaS. But I guess we're doing hosted control plane first?

At least in the foreseeable future, supporting the self-hosted path is not a priority afaik, but @derekwaynecarr can likely provide more insight into that.